File: WEB-INF/cgi-bin/env.py
#!/usr/bin/python3

import os

print("Content-type: text/html\n\n")
print("<html>")
print("<head>")
print("<title>Environment Variables</title>")
print("</head>")
print("<body>")
print("<h1>Environment Variables</h1>")
print("<table>")
print("<tr><th>Variable</th><th>Value</th></tr>")
for var in os.environ:
    value = os.environ[var]
    print("<tr><td>{}</td><td>{}</td></tr>".format(var, value))
print("</table>")
print("</body>")
print("</html>")

File: WEB-INF/cgi-bin/env.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

extern char** environ;

int main() {
    const char s[2] = "=";

    printf("Content-type: text/html\n\n");
    printf("<html><head><title>Environment Variables</title></head><body>\n");
    printf("<h1>Environment Variables</h1>\n");
    printf("<table border='1'><tr><th>Variable</th><th>Value</th></tr>\n");
    
    char **env = environ;
    while (*env) {
        char *env_var = *env;
        char *value = getenv(*env);
        char * token = strtok(*env, s);
        printf("<tr><td>%s</td><td>%s</td></tr>\n", token, value);
        env++;
    }

    printf("</table></body></html>\n");

    return 0;
}